lib/switchtec.c

```diff
@@ -56,6 +57,35 @@
  * @{
  */
 
+/* maximum firmware module string length */
+#define FW_MOD_NAME_LEN_MAX	32
+
+/* maximum number of log entries for each module */
+#define LOG_ENTRIES_MAX		512
+
+/* maximum log entry string length */
+#define LOG_ENTRY_LEN_MAX	512
+
+/* initial number of log definition modules to allocate */
+#define INIT_ALLOC_LOG_MODULES	200
+
+/**
+ * @brief Module-specific log definitions
+ */
+struct module_log_defs {
+	char mod_name[FW_MOD_NAME_LEN_MAX];	//!< module name
```

Comment:
I'd still use strdup for mod name (more on that below)...

Reply:
Sure, I'll use strdup. I didn't bother since our module names will never be more than around 12 characters, and malloc block sizes won't be less than 32 bytes anyway.

Reply:
The point is to get rid of the ugly global define that doesn't need to be there and the ugly %31s (where the 31 should actually be the global define).